acceptance/tests/sameness/sameness_test.go

```diff
@@ -776,3 +808,39 @@ func getPeeringAcceptorSecret(t *testing.T, cfg *config.TestConfig, server *clus
 
 	return acceptorSecretName
 }
+
+// localityForRegion returns the full api.Locality to use in tests for a given region string.
+func localityForRegion(r string) api.Locality {
+	return api.Locality{
+		Region: r,
+		Zone:   r + "a",
+	}
+}
+
+// checkLocalities checks the given cluster for client and server service instances matching the locality expected
+// for the cluster.
+func checkLocalities(t *testing.T, c *cluster) {
+	for ns, svcs := range map[string][]string{
+		staticClientNamespace: {
+			staticClientName,
+			staticClientName + "-sidecar-proxy",
+		},
+		staticServerNamespace: {
+			staticServerName,
+			staticServerName + "-sidecar-proxy",
+		},
+	} {
+		for _, svc := range svcs {
+			cs := getCatalogService(t, c, svc, ns, c.partition)
+			assert.NotNil(t, cs.ServiceLocality, "service %s in %s did not have locality set", svc, c.name)
+			assert.Equal(t, c.locality, *cs.ServiceLocality, "locality for service %s in %s did not match expected", svc, c.name)
+		}
+	}
+}
+
+func getCatalogService(t *testing.T, c *cluster, svc, ns, partition string) *api.CatalogService {
+	resp, _, err := c.client.Catalog().Service(svc, "", &api.QueryOptions{Namespace: ns, Partition: partition})
```

Comment:
I wonder if we should add a retry here, just in case?

Reply:
Could do - I'm a little wary of peppering things w/ retries we don't necessarily need, though I get the desire to keep tests as stable as possible. I didn't here bc we have a good number of other non-retried service fetches from the catalog (including in this test for DNS setup), so I figured it was likely to be as stable as those.

Reply:
@wilkermichael if you're ok w/ it, I might hold off on this and if we see a general need to wrap catalog fetches in this and other tests, do it all at once... LMK if you have different thoughts

Reply:
Sound good